conans/test/integration/command/create_test.py

```diff
@@ -659,3 +657,30 @@ def generate(self):
         assert "FOO LIBDIRS: ['lib']" in client.out
         assert "FOO INCLUDEDIRS: ['include']" in client.out
         assert "FOO RESDIRS: ['res']" in client.out
+
+
+def test_create_keep_build_sets_generators_folder_even_with_cmake_layout_defined():
+    """
+    Issue related: https://github.com/conan-io/conan/issues/11785
```

Comment:
```suggestion
    Issue related: https://github.com/conan-io/conan/issues/11785
    We need the `self.generators_folder` available because the CMake(self).install needs access to that folder to 
    read from the presets the "cmake generator" used.
```